clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -358,6 +452,15 @@ public int sessionId() {
                 .setResponses(topicResponseList);
     }
 
+    private Boolean supportsTopicIds() {
+        return data.responses().stream().findFirst().filter(
+            topic -> !topic.topicId().equals(Uuid.ZERO_UUID)).isPresent();
+    }
+
+    public Set<Uuid> topicIds() {
+        return data.responses().stream().map(resp -> resp.topicId()).filter(id -> !id.equals(Uuid.ZERO_UUID)).collect(Collectors.toSet());
```

Comment:
This suggests we can have a combination of zero and non-zero?

Reply:
I think this was a mistake. I need to see why I wrote it this way.

Reply:
I remember why I did this. I wanted to not get a set of the zero ID when the version was old. I think if we are able to get better versioning logic, this should be fixed easily.